From 22c6f42dcf951522d5bc77952c49064daf35f346 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:43 -0400
Subject: [scsi] qla2xxx: Decrease pci access for response queue processing for ISPFX00

Message-id: <1408391817-6389-56-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87965
O-Subject: [PATCH RHEL 7.1 e-stor 55/69] qla2xxx: Decrease pci access for response queue processing for ISPFX00.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Armen Baloyan <armen.baloyan@qlogic.com>

scsi commit id 9929c478123d3ce6020c5b771c8e8524ebb636fb

Signed-off-by: Armen Baloyan <armen.baloyan@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index 4104083..4bbeb63 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -2677,7 +2677,7 @@ qlafx00_process_response_queue(struct scsi_qla_host *vha,
  uint16_t lreq_q_out = 0;
 
  lreq_q_in = RD_REG_DWORD(rsp->rsp_q_in);
- lreq_q_out = RD_REG_DWORD(rsp->rsp_q_out);
+ lreq_q_out = rsp->ring_index;
 
  while (lreq_q_in != lreq_q_out) {
   lptr = rsp->ring_ptr;
-- 
1.7.1